tests/ui/option_and_then_some.rs

```diff
@@ -0,0 +1,16 @@
+// run-rustfix
+#![deny(clippy::option_and_then_some)]
+
+// need a main anyway, use it get rid of unused warnings too
+pub fn main() {
+    let x = Some(5);
+    // the easiest cases
+    x.and_then(Some);
```

Comment:
This is a noop, can we suggest `x` in this case?

Reply:
Done.